### PR TITLE
Replace request-waiting logic to use asyncio.Future

### PR DESCRIPTION
**Pull request by tsunyoku**
This PR provides more efficient & reliable logic for matching requests and responses.

### CODE REVIEW

File: surrealdb/clients/ws.py [hunk 1]
```diff
@@ -58,7 +59,7 @@ def __init__(
         self._ws: ClientWebSocketResponse
         self._recv_task: asyncio.Task
 
-        self._responses: dict[str, RPCResponse] = {}
+        self._response_futures: dict[str, asyncio.Future] = {}
```
Comment (CaedenPH) [inline surrealdb/clients/ws.py:63]: In the imports, include
`from typing import Dict`

```suggestion
        self._response_futures: Dict[str, asyncio.Future] = {}
```

**Review by CaedenPH — COMMENTED** [review]
To keep consistency, you should import `Dict` from` typing`

File: surrealdb/clients/ws.py (hunk 1 above)
Comment (tsunyoku) [reply surrealdb/clients/ws.py:63]: oops

**Review by CaedenPH — APPROVED** [review]
👍

**Pull request merged**